<launch>

  <arg name="ns" default="track_anything" />
  <arg name="model_type" default="vit_b" />
  <arg name="device" default="cuda:0" />
  <arg name="input_image" default="/kinect_head/rgb/image_rect_color"/>
  <arg name="input_depth" default="/kinect_head/depth_registered/image"/>
  <arg name="input_camera_info" default="/kinect_head/depth_registered/camera_info"/>
  <arg name="processed_image" default="processed_image_color"/>
  <arg name="compressed" default="false" />
  <arg name="input_point" default="processed_image_color/screenpoint" />
  <arg name="output_image" default="segmentation_image" />
  <arg name="color_fix_image" default="color_fix_image/compressed" />
  <arg name="segmentation_mask" default="segmentation_mask" />
  <arg name="width" default="640"/>
  <arg name="height" default="480"/>



  <arg name="_input_image" default="/$(arg namespace)/decompressed_image" if="$(arg compressed)"/>
  <arg name="_input_image" default="$(arg input_image)" unless="$(arg compressed)"/>


  <include file="$(find track_anything_ros)/launch/include/resize.launch" >
    <arg name="width" value="$(arg width)" />
    <arg name="height" value="$(arg height)" />
    <arg name="ns" value="$(arg ns)" />
    <arg name="input" value="$(arg input_image)" />
  </include>

  <group ns='$(arg ns)'>
    <node name="tracking_node"
          pkg="track_anything_ros" type="track_test.py"
          output="screen" >
      <remap from="~input_image" to="$(arg processed_image)" />
      <remap from="~input_point" to="$(arg input_point)" />
      <remap from="~output_image" to="$(arg output_image)" />
      <remap from="~color_fix_image" to="$(arg color_fix_image)" />
      <remap from="~segmentation_mask" to="$(arg segmentation_mask)" />
      <rosparam subst_value="true" >
        device: $(arg device)
        model_type: $(arg model_type)
        model_dir: $(find track_anything_ros)/checkpoints/
        tracker_config_file: $(find track_anything_ros)/config/tracker_config.yaml
        inpainter_config_file: $(find track_anything_ros)/config/inpainter_config.yaml
      </rosparam>
    </node>


    <!-- <include file="$(find track_anything_ros)/launch/decompress_depth.launch"> -->
    <!--   <arg name="input_image" value="$(arg input_image)"/> -->
    <!--   <arg name="input_depth" value="$(arg input_depth)"/> -->
    <!-- </include> -->

    <node pkg="jsk_rqt_plugins" type="rqt_image_view2" name="prompter"
          output="screen">
      <remap from="event" to="image_view2/event" />
      <remap from="image_marked" to="image_marked" />
    </node>
    <node pkg="image_view2" type="image_view2" name="image_view2"
          output="screen">
      <remap from="image" to="$(arg processed_image)" />
      <param name="use_window" value="false" />
    </node>

    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="$(arg output_image)" />
    </node>

    <node name="track_anything_button"
          pkg="jsk_rqt_plugins" type="rqt_tabbed_buttons" output="screen" clear_params="true">
      <rosparam command="load" file="$(find track_anything_ros)/resource/remote_tabbed_layout.yaml" />
    </node>
  </group>
</launch>
